Adicionando 1263 (#150)
petbccufscar/URI: Strings/1263/README.md

# Problema:    
Uma aliteração ocorre quando duas ou mais palavras consecutivas de um texto possuem a mesma letra inicial (ignorando maiúsculas e minúsculas). Sua tarefa é desenvolver um programa que identifique, a partir de uma sequência de palavras, o número de aliterações que essa sequência possui.

Entrada<br/>

> A entrada contém diversos casos de testes. Cada caso é expresso como um texto em uma única linha, contendo de 1 a 100 palavras separadas por um único espaço, cada palavra tendo de 1 a 50 letras minúsculas ou maiúsculas ('A'-'Z','a'-'z'). A entrada termina em EOF.ificada no exercício [exemplo da entrada/saída] (...)”. Caso contrário, deve ser omitida.

Reviewer: Acho que está errado depois de "A entrada termina em EOF", mas não sei o que era para estar escrito

```c
char frase[5099];
int espaco, inicial, count, ehSequencia; 
```

Para começar a resolução, precisamos definir um loop para ler todos os casos de teste. Fazemos então um `while` que fica em loop lendo a entrada até que ela seja EOF. Utilizamos o `fgets` para ler a entrada pois ele lê a linha toda, enquanto o usual `scanf` lê até o primeiro espaço, então pegaria somente a primeira palavra. Para utilizar o `fgets` precisamos passar a variável que vai armazenar a entrada, o tamanho máximo da entrada e da onde essa entrada será lida (no caso `stdin` que é o teclado). A condição do `while` funciona pois quando o `fgets` receber um EOF, será retornado um `NULL` então o loop irá acabar. 

```c
while(fgets(frase, sizeof(frase), stdin) != NULL){ 
```

> #### Para aprender um pouco mais sobre fgets: [Fgets](http://www.w3big.com/pt/cprogramming/c-function-fgets.html)
> <br/>

Reviewer: ```suggestion
```

Dentro do `while` iniciamos nossas variáveis com o valor 0. Pois a primeira inicial é 0, nosso contador também deve ser iniciado em 0 e ainda não encontramos uma sequência então ehSequencia também é 0. 

```c
inicial = 0; 
count = 0;
ehSequencia = 0;
```

> Então iteramos pela string com um `for` que vai de 0 até o tamanho da string, para obter o tamanho da string, utilizamos a função [`strlen`](http://linguagemc.com.br/a-biblioteca-string-h/) da biblioteca `<string.h>`. Então temos um `if` para perguntar se aquele caractere é um espaço ou não, se ele for um espaço, sabemos que o próximo caracter (espaco+1) é a inicial da próxima palavra. Então comparamos a inicial da primeira palavra com a próxima, utilizamos a função [`tolower`](http://linguagemc.com.br/ctype-h-toupper-tolower-isalpha-isdigit-em-c/) (que é definida na biblioteca `<ctype.h>`) para os dois caracteres ficarem em caixa baixa, justamente para não haver distinção entre letras minúsculas e maiúsculas. Se as duas iniciais forem iguais, verificamos se estamos ou não em uma sequência com mais um `if`, se não estamos em uma sequência, então isso é uma aliteração, então adicionamos um no nosso contador e definimos ehSequencia = 1, pois agora estamos em uma sequência de iniciais consecutivas iguais. 

Reviewer: ```suggestion
Em seguida, iteramos pela string com um `for` que vai de 0 até o tamanho da string, para obter o tamanho da string, utilizamos a função [`strlen`](http://linguagemc.com.br/a-biblioteca-string-h/) da biblioteca `<string.h>`. Depois, temos um `if` para verificar se aquele caractere é um espaço ou não, se ele for um espaço, sabemos que o próximo caracter (espaco+1) é a inicial da próxima palavra. Então, comparamos a inicial da primeira palavra com a próxima, utilizamos a função [`tolower`](http://linguagemc.com.br/ctype-h-toupper-tolower-isalpha-isdigit-em-c/) (que é definida na biblioteca `<ctype.h>`) para os dois caracteres ficarem em caixa baixa, justamente para não haver distinção entre letras minúsculas e maiúsculas. Se as duas iniciais forem iguais, verificamos se estamos ou não em uma sequência com mais um `if`, se não estamos em uma sequência, isso é uma aliteração, portanto adicionamos um no nosso contador e definimos `ehSequencia = 1`, pois agora estamos em uma sequência de iniciais consecutivas iguais. 
```

```c
for(espaco = 0; espaco < strlen(frase); espaco++)
    if(frase[espaco] == ' '){
        if(tolower(frase[inicial]) == tolower(frase[espaco+1])){
            if(!ehSequencia){
                count++;
                ehSequencia = 1;
            }
        }
```

> Se as iniciais não forem iguais, ou seja se `if(tolower(frase[inicial]) == tolower(frase[espaco+1]))` retornar falso, então não estamos em uma sequência iniciais iguais, então fazemos `ehSequencia = 0`. Note que a variável `ehSequencia` funciona como uma variável booleana, 0 para falso e 1 para verdadeiro. Essa variável serve justamente para não contarmos aliterações a mais, pois podemos ter uma frase como por exemplo: "four fanatic fantastic fans" em que todas as iniciais são iguais, porém só há uma aliteração. Por isso contamos uma aliteração quando as iniciais são iguais e elas não estão em sequência. Por fim, fazemos com que a próxima inicial seja o `espaco+1` pois na próxima iteração vamos comparar com a palavra seguinte. 

Reviewer: ```suggestion
Se as iniciais não forem iguais, ou seja se `if(tolower(frase[inicial]) == tolower(frase[espaco+1]))` retornar falso, então não estamos em uma sequência iniciais iguais, portanto `ehSequencia = 0`. Note que a variável `ehSequencia` funciona como uma variável booleana, 0 para falso e 1 para verdadeiro. Essa variável serve justamente para não contarmos aliterações a mais, pois podemos ter uma frase como por exemplo: "four fanatic fantastic fans" em que todas as iniciais são iguais, porém só há uma aliteração. Por isso, contamos uma aliteração quando as iniciais são iguais e não estão em sequência. Por fim, fazemos com que a próxima inicial seja o `espaco+1` , pois na próxima iteração vamos comparar com a palavra seguinte. 
```

```c
    else
        ehSequencia = 0;

    inicial = espaco+1;
}
```

Finalmente, apresentamos o resultado fazendo um `printf` do count.

```c
printf("%d\n", count); 
```

>     

Reviewer: ```suggestion

#### Para aprender um pouco mais sobre fgets: [Fgets](http://www.w3big.com/pt/cprogramming/c-function-fgets.html)

```
